api/mesh/v1alpha1/dataplane_helpers.go

```diff
@@ -269,6 +278,15 @@ func (d *Dataplane) Tags() MultiValueTagSet {
 			tags[tag][value] = true
 		}
 	}
+	if d.GetNetworking().GetGateway() != nil {
```

Comment:
I think, `if` is not necessary of we use `d.GetNetworking().GetGateway().GetTags()` in the `for` loop